Add Spinnaker scripts

ejizba opened the pull request.

ejizba committed:
Copy Spinnaker files over from quickstart template

ejizba committed:
Refactor spinnaker scripts to be user-friendly

Add better parameter handling
Add a README for each script
Split out some sections of the 'set_spinnaker' script

msftclas commented:
@EricJizbaMSFT,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

clguiman commented on quickstart_template/spinnaker_vm_to_kubernetes.sh:
```diff
@@ -0,0 +1,73 @@
+
+# Create pipeline if enabled
+if (( $include_kubernetes_pipeline )); then
+    if [ "$pipeline_registry" == "$docker_hub_registry" ]; then
```
why not just pass the registry url as an argument? Then you don't need the extra check

clguiman commented on spinnaker/add_k8s_pipeline/README.md:
```diff
@@ -0,0 +1,31 @@
+|---|---|
+| --account_name<br/>-an | The Spinnaker account name for your registry |
+| --registry<br/>-rg | The Azure Container Registry url, for example 'sample-microsoft.azurecr.io'. |
+| --repository<br/>-rp | The repository, for example 'Farikam/app1'. Any new tag pushed to this repository will trigger the pipeline. |
```
typo. Fabrikam instead of Farikam :)

clguiman commented on spinnaker/await_restart_spinnaker/await_restart_spinnaker.sh:
```diff
@@ -0,0 +1,51 @@
+# Front50 - port 8080
+# Orca - port 8083
+count=0
+while !(nc -z localhost 8080) || !(nc -z localhost 8084) || !(nc -z localhost 7002) || !(nc -z localhost 8083); do
```
is it ok to hardcode ports?

ejizba replied:
There's a very small chance the user changes their ports and tries to use this script and fails. It might be possible to dynamically check which port maps to which service, but I don't think the effort is worth it at this point

clguiman commented on spinnaker/copy_kube_config/README.md:
```diff
@@ -0,0 +1,25 @@
+# Copy kubeconfig file
+
+Progamatically copies a kubeconfig file from an Azure Container Service Kubernetes cluster to a Spinnaker machine.
```
typo

clguiman commented on spinnaker/copy_kube_config/remove_temp_user.sh:
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
I don't think this script really makes sense, it's a one-liner. Do you plan to add extra commands in the future?

ejizba replied:
Refactored to use VMAccesLinuxExtention so that this script and add_temp_user script are no longer necessary. I also refactored to use python JSON parsing when getting the nameSuffix to make it more robust. @clguimanMSFT PTAL at these refactorings

clguiman reviewed (approved):
Looks good. I just added some observations

ejizba committed:
Fix review comments

Biggest changes:
Refactor to use VMAccessForLinux
Refactor to use python json parsing

ejizba committed:
Fix application failure when email is empty

This parameter isn't _required_ because it has a default, but it also can't be empty. So I refactored the messaging.
